### PR TITLE
fib: make gateway prefix check more robust

### PR DESCRIPTION
**Pull request by cjdoucette**

### CODE REVIEW

File: gk/fib.c [hunk 1]
```diff
@@ -1371,10 +1372,19 @@ add_fib_entry_locked(struct ip_prefix *prefix,
...
+/*
+ * Return 0 when @gw_addr is not included in @prefix.
+ * If not, or if there is an error, return -1.
+ */
 static int
 check_gateway_prefix(struct ip_prefix *prefix, struct ipaddr *gw_addr)
 {
+	if (unlikely(prefix->addr.proto != gw_addr->proto)) {
+		GK_LOG(ERR, "IP prefix protocol (%hu) does not match the gateway address protocol (%hu)\n",
+			prefix->addr.proto, gw_addr->proto);
```
Comment (AltraMayor) [inline gk/fib.c:1384]: Add `prefix->str` to the log entry.

File: gk/fib.c [hunk 2]
```diff
@@ -1400,9 +1410,10 @@ check_gateway_prefix(struct ip_prefix *prefix, struct ipaddr *gw_addr)
 					((pf[1] ^ gw[1]) & ip6_mask))
 				return 0;
 		}
-	} else
-		rte_panic("Unexpected condition at %s: unknown IP type %hu\n",
+	} else {
+		GK_LOG(ERR, "Unexpected condition at %s: unknown IP type %hu\n",
 			__func__, gw_addr->proto);
+	}
```
Comment (AltraMayor) [inline gk/fib.c:1416]: Add `prefix->str` to the log entry.

**Review by AltraMayor — CHANGES_REQUESTED** [review]
**fib: make gateway prefix check more robust**

**Comment by cjdoucette** [comment]
Ready for another review.

**Pull request merged**